### ISSUE TITLE
[cmd/mdatagen] Ability to disable/enable all metrics

### ISSUE THREAD

**Issue by dmitryax** [open-telemetry/opentelemetry-collector#10074]
### Problem

We have a few issues when it's not desirable to configure the emitted metrics and resources starting from the "default" set. Sometimes, it's needed to start with an empty set and enable just a few metrics or start with enabling all metrics/resources (default+optional) and explicitly disabling a few of them. 

This is needed at least for:
- https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28832
- https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30894

### Possible solutions

Given that the `metrics` and `resource_attributes` interfaces provide a map `name: config`, we need to introduce special metric names that can be applied to a group of metrics/resources.

#### Option 1

Have a predefined special metric name that cannot collide with existing names, e.g., `_ALL`.

```yaml
metrics:
  _ALL:
    enabled: false
  metric1:
    enabled: true
```

#### Option 2

Support glob-style `*` in metric names. Using `*` in metric/attribute names isn't valid according to the OTel specification, so it cannot cause collisions. Providing this capability makes the user interface more powerful, as users will be able to disable/enable groups of metrics by their namespaces.

```yaml
metrics:
  '*':
     enabled: false
  metric1:
    enabled: true
```

Providing this interface can cause collisions if the several globs match the same metric (e.g. `*` and `system.*` match any `system.*` metric). Once such collisions are detected, the config validation should fail because there is no way to clearly define precedence with a map.

This approach requires an evaluation from the implementation perspective. It might over-complicate the mdatagen logic.

**Comment by dmitryax** [comment]
@braydonk, I don't think we can rely on the order of the map keys because it's not supposed to be preserved

**Comment by braydonk** [comment]
The PR I posted takes that into account and works regardless of the order. There's a custom priority system. The priority of the pattern goes down for every `.` in the name, so `*` will be viewed as higher priority than `x.*` regardless of what order the patterns are read.

**Comment by dmitryax** [comment]
How does it work with the `{...}`? Which one wins compared to `*`? What if I have `prefix.*` and `*.suffix`?

**Comment by dmitryax** [comment]
Thank you for putting the doc! Once we have the PR merged, please add push a PR to add this in `cmd/mdatagen`

**Comment by jehon** [comment]
Any news on this? I would love to have something like that.
My use case: I only look at some metrics, so I want to disable all except those I look at.

**Comment by dmitryax** [comment]
This is still a desired feature, but we haven’t had a chance to implement it yet. The wildcard proposal seemed overly complicated, and we wanted something simpler. I found [this comment](https://github.com/open-telemetry/opentelemetry-collector/pull/13358#issuecomment-3113491968), but I don’t remember the specifics of what was discussed.

@braydonk  do you recall any details by chance?

**Comment by braydonk** [comment]
We can consider my RFC completely defunct. I heavily overcomplicated how the wildcard patterns would work to allow the wildcards to be used directly in yaml config, but it was really not worth the complexity.

As a new approach, I have opened #15652 to introduce a new type of pattern matching to the `filter` package. If that gets merged, then we could create configs within the generated `MetricsBuilder` config to use that filter and enable mass enable/disable using wildcard matching. Please take a look if you are interested.
